scripts/publish-deb.sh

```diff
@@ -73,12 +67,32 @@ do
 
   DEBS3_SHOW="deb-s3 show $BUCKET_ARG $S3_REGION_ARG"
 
-  deb_split=(${deb//_/ })
-  deb="${deb_split[0]}"
-  deb=$(basename $deb)
+  # extracting name from debian package path. E.g:
+  # _build/mina-archive_3.0.1-develop-a2a872a.deb -> mina-archive
+  deb=$(basename "$deb")
+  deb="${deb%_*}"
+  
   
-  for i in {1..10}; do (verify_o1test_repo_has_package $deb) && break || sleep 60; done
+  for i in {1..10}; do 
 
+    sudo apt-get update
+    ${DEBS3_SHOW} "$deb" "${DEB_VERSION}" "${ARCH}" -c "${DEB_CODENAME}" -m "${DEB_RELEASE}"
+    LAST_VERIFY_STATUS=$?
+    
+    if [[ $LAST_VERIFY_STATUS == 0 ]]; then
+        echo "succesfully validated that package is uploaded to deb-s3"
+        break
+    fi
+    
+    sleep 60
```

Comment:
The worst case of this is still pathological (`num_debs * 10m`) and I'd like to see the loop inverted (try all debs, if any fail then wait 60s, 10 attempts).

Reply:
In some ancient times i tried approach like you mentioned above. However, another issue pops out. Single job locked debian repository for too long (pushing all debs in one go takes couple of minutes ) and other publishing jobs were failing somtimes causing debian repo corruption. So in general, small updates are better from my perspective . In order to make it successful probably we would need to either expand wait time for lock release or upload just a one debian at time, release lock and then go to next one. 

In the future i would like to remove publish debians on PR as we are not using them in tests nor in docker builds (we are relying on gcp bucket and local repo). This will greatly decrease number of parties which pushes debians and then probably your approach would be ideal

Reply:
This isn't pushing any debian packages though. This is just checking them. We don't need to wait 10 minutes for each debian file, we could check them all at once.